src/drivers/rest/data.rs

```diff
@@ -0,0 +1,164 @@
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+
+use lazy_static::lazy_static;
+use mavlink::{self, Message};
+use serde::{Deserialize, Serialize};
+
+lazy_static! {
+    static ref DATA: Data = Data {
+        messages: Arc::new(Mutex::new(MAVLinkVehiclesData::default())),
+    };
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct MAVLinkMessage<T> {
+    pub header: mavlink::MavHeader,
+    pub message: T,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct MAVLinkVehiclesData {
+    vehicles: HashMap<u8, MAVLinkVehicleData>,
+}
+
+impl MAVLinkVehiclesData {
+    fn update(&mut self, message: MAVLinkMessage<mavlink::ardupilotmega::MavMessage>) {
+        let vehicle_id = message.header.system_id;
+        self.vehicles
+            .entry(vehicle_id)
+            .or_insert(MAVLinkVehicleData {
+                id: vehicle_id,
+                components: HashMap::new(),
+            })
+            .update(&message);
+    }
+
+    pub fn pointer(&self, path: &str) -> String {
```

Comment:
Shouldn't this be private?
```suggestion
    fn pointer(&self, path: &str) -> String {
```

Reply:
done